src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractRunCodeAnalysisCommand.cs

```diff
@@ -0,0 +1,203 @@
+
+        protected abstract string GetCommandText();
+        protected abstract bool ShouldHandle(IProjectTree node);
+        protected abstract long CommandId { get; }
+
+        protected async override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string commandText, CommandStatus progressiveStatus)
+        {
+            if (ShouldHandle(node))
+            {
+                // Enable the command if code analysis is enabled and the build manager is ready to build.
+                if (await IsCodeAnalysisPackageInstalledAsync().ConfigureAwait(false))
+                {
+                    var commandStatus = await IsReadyToBuildAsync().ConfigureAwait(false) ? CommandStatus.Enabled : CommandStatus.Supported;
+                    return await GetCommandStatusResult.Handled(GetCommandText(), commandStatus).ConfigureAwait(false);
+                }
+             }
+
+            return CommandStatusResult.Unhandled;
+        }
+
+        private async Task<bool> IsCodeAnalysisPackageInstalledAsync()
+        {
+            if (!_isCodeAnalysisPackageInstalled.HasValue)
+            {
+                // Switch to UI thread for querying the build manager service.
+                await _threadingService.SwitchToUIThread();
+
+                if (_vsShell == null)
+                {
+                    _vsShell = _serviceProvider.GetService<IVsShell, SVsShell>();
+                }
+
+                var codeAnalysisGuid = new Guid(ManagedProjectSystemPackage.CodeAnalysisPackageGuid);
+                ErrorHandler.ThrowOnFailure(_vsShell.IsPackageInstalled(ref codeAnalysisGuid, out int packageInstalled));
+                _isCodeAnalysisPackageInstalled = packageInstalled != 0;
+            }
+
+            return _isCodeAnalysisPackageInstalled.Value;
+        }
+
+        private async Task EnsureCodeAnalysisPackageLoadedAsync()
+        {
+            if (!_codeAnalysisPackageLoadAttempted)
+            {
+                // Switch to UI thread for querying the build manager service.
+                await _threadingService.SwitchToUIThread();
+
+                if (_vsShell == null)
+                {
+                    _vsShell = _serviceProvider.GetService<IVsShell, SVsShell>();
+                }
+
+                var codeAnalysisGuid = new Guid(ManagedProjectSystemPackage.CodeAnalysisPackageGuid);
+                _vsShell.IsPackageLoaded(ref codeAnalysisGuid, out IVsPackage package);
+                if (package == null)
+                {
+                    _vsShell.LoadPackage(ref codeAnalysisGuid, out package);
+                }
+
+                _codeAnalysisCommandTarget = package as IOleCommandTarget;
+                _codeAnalysisPackageLoadAttempted = true;
+            }
+        }
+
+        private async Task<bool> IsReadyToBuildAsync()
+        {
+            // Ensure build manager is initialized.
+            await EnsureBuildManagerInitializedAsync().ConfigureAwait(true);
```

Comment:
Why the `ConfigureAwait(true)` here? I don't believe you're coming from a specific thread on this.